### PR TITLE
Set --enable-scenarios to False by default

### PR DESCRIPTION
**Pull request by akrmn**
Let's see ~if anything~ what breaks!

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs [hunk 1]
```diff
@@ -172,10 +172,11 @@ getIntegrationTests registerTODO scenarioService = do
     putStrLn $ "rtsSupportsBoundThreads: " ++ show rtsSupportsBoundThreads
     do n <- getNumCapabilities; putStrLn $ "getNumCapabilities: " ++ show n
 
-    damlTests <- getDamlTestFiles "compiler/damlc/tests/daml-test-files"
-    bondTradingTests <- getBondTradingTestFiles
-    let plainTests = damlTests <> bondTradingTests
-    noScenariosEnabledTests <- getDamlTestFiles "compiler/damlc/tests/daml-test-files/no-scenarios-enabled"
+    plainTests <- getDamlTestFiles "compiler/damlc/tests/daml-test-files"
```
Comment (akrmn) [inline compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs:175]: I also flip the tests. Now, `plainTests` refers to the tests that run with `--enable-scenarios=no`, and `scenariosEnabledTests` (including the bond trading test) are the "marked" case, running with the non-default `--enable-scenarios=yes`. I guess with time we should try to move tests from the latter to the former, or at least avoid adding new tests to the latter.

**Comment by akrmn** [comment]
Hm, the first test to fail was `//compiler/damlc/tests:daml-doc`, because it didn't expect the `daml-test-files` to be nested in another directory. 

I think it's easier to change the directory approach than to fix the daml-doc test, so I switched to a new test annotation `-- @ENABLE-SCENARIOS`. This requires reading the annotations before the actual test, but I think that should be fine. Note that I added the annotations (when needed) at the _end_ of the file, rather than at the top as usual, since otherwise I'd have to adjust all the expected ranges for `-- @WARN` and `-- @ERROR` annotations

**Review by sofiafaro-da — APPROVED** [review]
Looks good to me, thank you!

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by akrmn** [comment]
not merging just yet, I feel like some tests without explicitly-enabled scenarios should be failing

**Comment by akrmn** [comment]
Found the problem: I forgot to update the default in `enableScenariosOpt`. Now we'll probably get the test failures we were expecting :)

File: compiler/damlc/tests/src/daml-ghc-deterministic.sh [hunk 2]
```diff
@@ -28,6 +28,7 @@ TESTS_DIR=$(dirname $(rlocation "$TEST_WORKSPACE/compiler/damlc/tests/daml-test-
 damlc=$(rlocation "$TEST_WORKSPACE/$1")
 protoc=$(rlocation "$TEST_WORKSPACE/$2")
 diff="$3"
+scenarios="--enable-scenarios=yes" # TODO: https://github.com/digital-asset/daml/issues/11316
```
Comment (cocreature) [inline compiler/damlc/tests/src/daml-ghc-deterministic.sh:31]: not sure why there is even a scenario in here, we should just use a module without one.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 3]
```diff
@@ -67,11 +67,15 @@ authorizationHeaders = [("Authorization", "Bearer " <> T.encodeUtf8 hardcodedTok
 
 withDamlServiceIn :: FilePath -> String -> [String] -> IO a -> IO a
 withDamlServiceIn path command args act = withDevNull $ \devNull -> do
-    let proc' = (shell $ unwords $ ["daml", command] <> args)
-          { std_out = UseHandle devNull
-          , create_group = True
-          , cwd = Just path
-          }
+    let proc' =
```
Comment (cocreature) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:70]: Given that the integration tests are more or less testing what we expect users to use, we should migrate those pre 2.0. I don’t know what scenarios we use here but hopefully just a matter of replacing a few scenarios by init scripts. Let’s do it in a separate PR.

File: language-support/hs/bindings/examples/group-chat/BUILD.bazel [hunk 4]
```diff
@@ -8,10 +8,12 @@ load("@os_info//:os_info.bzl", "is_windows")
 daml_compile(
     name = "GroupChat",
     srcs = glob(["daml/GroupChat.daml"]),
+    enable_scenarios = True,  # TODO: https://github.com/digital-asset/daml/issues/11316
```
Comment (cocreature) [inline language-support/hs/bindings/examples/group-chat/BUILD.bazel:11]: while they’re called examples the HS bindings are only used internally so can be post 2.0

File: language-support/java/codegen/BUILD.bazel [hunk 5]
```diff
@@ -216,6 +216,7 @@ daml_build_test(
         ":pkg1.dar": "pkg1.dar",
         ":pkg2.dar": "pkg2.dar",
     },
+    enable_scenarios = True,
```
Comment (cocreature) [inline language-support/java/codegen/BUILD.bazel:219]: haven’t checked in detail but maybe we can just not have scenarios in here.

File: compiler/damlc/tests/src/daml-ghc-deterministic.sh (hunk 2 above)
Comment (akrmn) [reply compiler/damlc/tests/src/daml-ghc-deterministic.sh:31]: ah that's fair, do you have one in mind?

Comment (cocreature) [reply compiler/damlc/tests/src/daml-ghc-deterministic.sh:31]: could take the `User.daml` file from create-daml-app for example.

File: language-support/java/codegen/BUILD.bazel (hunk 5 above)
Comment (akrmn) [reply language-support/java/codegen/BUILD.bazel:219]: yeah, this one just has 
```
scenario = do
  p <- getParty "p"
  submit p $ create (V1.A.T1 p)
  submit p $ create (V2.A.T2 p)
```
I think that should work as a `Script`, right? I'll do it in a separate PR though

Comment (cocreature) [reply language-support/java/codegen/BUILD.bazel:219]: You can do it as a script but I’m not sure why we need it at all. I think we can just delete it and the tests should still work. But definitely  happy to do it in a separate PR.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 3 above)
Comment (akrmn) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:70]: yes, this one has a `TODO`

File: docs/BUILD.bazel [hunk 6]
```diff
@@ -571,6 +571,7 @@ java_binary(
 daml_test(
     name = "ledger-api-daml-test",
     srcs = glob(["source/app-dev/code-snippets/**/*.daml"]),
+    enable_scenarios = True,
```
[the remark this answers is not in the document]
Comment (akrmn) [reply docs/BUILD.bazel:574]: thanks, I've added a `TODO` for this one

**Comment by akrmn** [comment]
I think the test failure was a fluke, it ran fine on windows and macOS, and also on an ad-hoc machine I just started, so I'll re-run the linux tests

**Comment by cocreature** [comment]
merged so it doesn’t conflict over the holidays
